src/pip/_internal/cache.py

```diff
@@ -87,23 +99,27 @@ def _get_candidates(self, link, package_name):
             raise
 
     def get_path_for_link(self, link):
+        # type: (Link) -> str
         """Return a directory to store cached items in for link.
         """
         raise NotImplementedError()
 
     def get(self, link, package_name):
+        # type: (Link, Optional[str]) -> Link
         """Returns a link to a cached item if it exists, otherwise returns the
         passed link.
         """
         raise NotImplementedError()
 
     def _link_for_candidate(self, link, candidate):
+        # type: (Link, str) -> Link
         root = self.get_path_for_link(link)
         path = os.path.join(root, candidate)
 
         return Link(path_to_url(path))
 
     def cleanup(self):
+        # type: (...) -> None
```

Comment:
```suggestion
        # type: () -> None
```

The `...` isn't needed here.